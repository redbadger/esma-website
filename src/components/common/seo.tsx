import React from "react";
import { Helmet } from "react-helmet";
import { useStaticQuery, graphql } from "gatsby";

const getSocialImageSrc = (image: { src; alt }, site) => {
  const relativeImagePath = image.src;
  return `https://${site.siteMetadata.hostName}${relativeImagePath}`;
};

function SEO({ description, lang, meta, title, image }: SeoProperties) {
  const { site, allFile } = useStaticQuery(
    graphql`
      query {
        site {
          siteMetadata {
            title
            description
            author
            hostName
          }
        }
        allFile(filter: { relativePath: { eq: "logo/short.svg" } }) {
          nodes {
            publicURL
          }
        }
      }
    `
  );

  const metaDescription = description || site.siteMetadata.description;
  if (!image || !image.src) {
    image = { src: allFile.nodes.publicURL, alt: "ESMA logo" };
  }
  const imageSrc = getSocialImageSrc(image, site);

  return (
    <Helmet
      htmlAttributes={{
        lang,
      }}
      title={title}
      titleTemplate={`%s | ${site.siteMetadata.title}`}
      meta={[
        {
          name: `description`,
          content: metaDescription,
        },
        {
          property: `og:title`,
          content: title,
        },
        {
          property: `og:description`,
          content: metaDescription,
        },
        {
          property: `og:type`,
          content: `website`,
        },
        {
          property: `og:image`,
          content: imageSrc,
        },
        {
          property: `og:image:alt`,
          content: image.alt,
        },
        {
          name: `twitter:card`,
          content: `summary`,
        },
        {
          name: `twitter:creator`,
          content: site.siteMetadata.author,
        },
        {
          name: `twitter:title`,
          content: title,
        },
        {
          name: `twitter:description`,
          content: metaDescription,
        },
        {
          name: "twitter:image",
          content: imageSrc,
        },
        {
          name: "twitter:image:alt",
          content: image.alt,
        },
      ].concat(meta)}
    />
  );
}

SEO.defaultProps = {
  lang: `en`,
  meta: [],
  description: ``,
};

interface SeoProperties {
  description: string;
  lang: string;
  meta: { name: string; content: string }[];
  title: string;
  image: {
    src: string;
    alt: string;
  };
}

export default SEO;
