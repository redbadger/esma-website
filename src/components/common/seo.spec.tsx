import React from "react";
import renderer from "react-test-renderer";
import { PureSeo } from "./seo";

describe("PureSeo", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <PureSeo
          description={"Tests for ESMA"}
          title={"Welcome to ESMA Test"}
          image={{ src: "/barry.png", alt: "A Barry" }}
          meta={[
            { property: "og:custom", content: "test content" },
            { name: "twitter:other", content: "some twitter metadatum" },
          ]}
          lang="en"
          allFile={{ nodes: [{ publicURL: "/path/to/logo.svg" }] }}
          site={{
            siteMetadata: {
              title: "title",
              author: "author",
              hostName: "hostName",
              description: "description",
            },
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
