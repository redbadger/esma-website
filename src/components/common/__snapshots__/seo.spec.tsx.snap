// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PureSeo renders correctly 1`] = `null`;
